src/icloudpd/authentication.py

```diff
@@ -50,6 +57,22 @@ def authenticate_(
     return authenticate_
 
 
+def request_2fa(icloud: pyicloud_ipd.PyiCloudService, logger: logging.Logger):
```

Comment:
This is new, and significant. Before this was here, the `request_2sa` codepath was executed, and I'd select that I wanted a 2fa sent to my devices vs an SMS message. I'd get the 2FA prompt on my laptop, and enter the 2FA code into the `icloudpd` prompt, but it would always fail to validate.

But, after I added this code, it seems to work as I'd expect. I enter my password, and get immediately prompted on my macos to validate the new device, and then given a 2FA code that I enter in to icloudpd, which then validates it and starts downloading photos/videos.

I do not completely understand the difference in API calls between 2sa and 2fa, but, it's significant enough to get it to pass one way (2fa) and fail the other (2sa). 🤷🏻